README.md

```diff
@@ -7,24 +7,29 @@ Elastic AI implementation for the Elastic Node version 5 (enV5).
 
 ## Setup
 
-For compiling the project the tools are:
+For compiling the project, the required tools are:
 
-- required:
-  - gcc (local C compiler)
-  - arm-non-eabi-gcc (C compiler for Target device)
-  - CMake (Build Script Generator)
-  - Ninja (Build Tool)
-- recommended:
-  - pre-commit
+- gcc (local C compiler)
+- arm-non-eabi-gcc (C compiler for RP2040)
+- CMake (Build Script Generator)
+- Ninja (Build Tool)
+
+We recommended pre-commit to check that the commit message align with the conventional commit mesage standard.
 
 > You can check if your local machine satisfies the required dependencies by executing `test_setup.sh` script
 > in the project [root directory](.).
 
+After pulling the repository, you are required to download all submodules (see [Submodules](#submodules)).
+This can be archived by executing 
+```bash
+git submodule update --init --recursive
```

Comment:
Of course. If we want to distribute a init script, yes. I would suggest adding an Issue, because to my point of view it's outside of the scope of this issue. Also the Documentation in general should be updated. That one the things i planned for myself in the next weeks. This would allow us to onboard new student workers without the large drawback of introducing them in length to our code base.